backend/src/app/services/site/site.service.ts

```diff
@@ -254,14 +258,16 @@ export class SiteService {
     }
 
     if (whenCreated) {
-      query.andWhere('sites.whenCreated = :whenCreated', {
-        whenCreated: whenCreated,
+      query.andWhere('sites.whenCreated BETWEEN :start AND :end', {
+        start: whenCreated[0],
+        end: whenCreated[1],
       });
     }
 
     if (whenUpdated) {
-      query.andWhere('sites.whenUpdated = :whenUpdated', {
-        whenUpdated: whenUpdated,
+      query.andWhere('sites.whenUpdated BETWEEN :start AND :end', {
+        start: whenUpdated[0],
```

Comment:
Please add  a validation for array length before accessing array index

Reply:
done: https://github.com/bcgov/nr-site-registry/pull/240/commits/2e69bc7eb0a0c127a5e28e51e16e06e3d7b9da68